---
title: Setup
---

## Platform and data access

This workshop is designed to be conducted on the [NeSI](https://www.nesi.org.nz) compute infrastructure. All software and data are already set up for you to use during the workshop.

{% comment %} Data is PFR's and unlikely to be made available for public use at this time, so not sure about adding info on running locally {% endcomment %} 

## Software

The  software needed to run this lesson are listed below. Versions reported are the ones we use on NeSI but the lesson is likely to work with a range of versions of these softwares.

| Software      | Version | Manual      | Description 	|
| ----------- | ----------- | ----------- | ----------- |
| Jellyfish | 2.3.0   | [link](https://github.com/gmarcais/Jellyfish) | K-mer analysis|
| FastQC    | 0.11.9  | [link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)| Illumina read QC |
| SeqKit    |         | [link](https://bioinf.shenwei.me/seqkit/) |||
| NanoStat  | 1.5.0   | [link](https://github.com/wdecoster/nanostat)| Nanopore read QC|
| nanofilt  | 2.6.0   |[link](https://github.com/wdecoster/nanofilt) |Nanopore read filtering |
| Flye      | 2.9.1-b1780 | [link](https://github.com/fenderglass/Flye) | Long-read genome assembler |
| BWA       | 0.7.17  | [link](https://bio-bwa.sourceforge.net/bwa.shtml) | Read mapping, assembly indexing |
| medaka    | 1.6.0   | [link](https://github.com/nanoporetech/medaka) | Assembly polishing |
| Pilon     | 1.24    | [link](https://github.com/broadinstitute/pilon/wiki)| Nanopore assembly polishing |
| SAMtools  | 1.15.1  | [link](http://www.htslib.org/doc/samtools.html) | Read mapping |
| QUAST     | 5.2.0   |[link](https://github.com/ablab/quast)| Assembly QC |
| BUSCO     | 5.1.3   | [link](https://busco.ezlab.org/)| Assembly QC - gene orthologs |
| KAT       | 2.4.2   | [link](https://kat.readthedocs.io/en/latest/) || K-mer based assembly QC | 
 


{% include links.md %}
